python/pyarrow/parquet/core.py

```diff
@@ -1803,6 +1803,9 @@ def __init__(self, path_or_paths, filesystem=None, schema=None,
             raise NotImplementedError("split_row_groups not yet implemented")
 
         if filters is not None:
+            import pyarrow.dataset as ds
+            if isinstance(filters, ds.Expression):
```

Comment:
I _think_ this might be unsafe, if dataset wasn't built. An alternative is to remove and just let it fail down in _check_filters, which will complete about `Expression` not having `len`.

Reply:
Just removed it in https://github.com/apache/arrow/pull/14011/commits/25dcf6af986cfc219a824ea36ea7c0ebb7e63aec.